src/jllvm/vm/Runtime.cpp

```diff
@@ -261,6 +262,59 @@ jllvm::Runtime::~Runtime()
 
 void jllvm::Runtime::prepare(ClassObject& classObject)
 {
+    // Lookup the JIT CC and interpreter CC implementations once ahead of time and save them in the method object.
+    // This is not only more convenient (allowing the method object to be called standalone), but is likely also faster
+    // on average as we do not have to do any repeated lookups and can batch the lookups here.
+    llvm::DenseMap<llvm::orc::SymbolStringPtr, Method*> methodMapping(classObject.getMethods().keys().size());
+    for (const Method& method : classObject.getMethods())
+    {
+        if (method.isAbstract())
+        {
+            // Abstract methods are not callable.
+            continue;
+        }
+        methodMapping.insert({m_interner(mangleDirectMethodCall(&method)), const_cast<Method*>(&method)});
+    }
+
+    // We perform the lookup asynchronously (purely because we can). Use two promises to ensure that both lookups are
+    // done prior to exiting the method.
+    std::promise<void> jitLookupDone;
```

Comment:
What is the advantage of doing this asynchronously?

Reply:
My original thought was mostly to batch-up symbol lookups as a super easy micro optimization. But doing so asynchronously is also very easy and in theory allows the method selection for the V-Table filling below to be done in parallel to all the symbol lookups. 
In practice we haven't enabled multithreading in LLVMs JIT yet so it currently runs synchronously in reality.